### PR TITLE
Don't read DNS responses until we get at least two bytes

### PR DESCRIPTION
**Pull request by zizhong**
Some name servers may send ATS empty TCP packages or 1-byte  packages. This patch is to hold the read until we get at least two bytes that indicate the size of the DNS response.

### CODE REVIEW

**Comment by maskit** [comment]
[approve ci]

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -808,6 +808,15 @@ DNSHandler::recv_dns(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
           dnsc->tcp_data.buf_ptr = make_ptr(dnsBufAllocator.alloc());
         }
         if (dnsc->tcp_data.total_length == 0) {
+          // see if TS gets a two-byte size
+          short tmp = 0;
```
Comment (maskit) [inline iocore/dns/DNS.cc]: It should be `uint16_t` because the size is important here.

Comment (zizhong) [reply iocore/dns/DNS.cc]: good idea! will fix.

**Comment by shinrich** [comment]
[approve ci]

**Comment by zizhong** [comment]
format fixed

**Comment by scw00** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
@zwoop is testing this patch on docs

**Pull request merged**